app/workers/irker_worker.rb

```diff
@@ -0,0 +1,145 @@
+require 'json'
+require 'socket'
+
+# Note: extra configuration possible, in config/gitlab.yml :
+# production:
+#   irker:
+#     server_ip: localhost
+#     server_port: 6659
+#     max_channels: 3
+#     # Next one has no default, this is just an example
+#     default_irc_uri: irc://chat.freenode.net/
+# You can place the 'irker' section just below the already-there 'extra' section
+
+class IrkerWorker
+  include Sidekiq::Worker
+
+  def perform(project_id, chans, colors, push_data)
+    project             = Project.find(project_id)
+
+    # Get config parameters
+    begin
+      irker_server = Gitlab.config.irker.server_ip
+      irker_port   = Gitlab.config.irker.server_port
+    rescue Settingslogic::MissingSetting
+      irker_server = 'localhost'
+      irker_port   = 6659
+    end
+
+    logger.info "IrkerWorker: messages destination = #{chans}"
+    return false unless setup_members chans, irker_server, irker_port
+
+    repo_name = push_data['repository']['name']
+    committer = push_data['user_name']
+    branch    = push_data['ref'].gsub(%r'refs/[^/]*/', '')
+
+    if colors
+      repo_name = "\x0304#{repo_name}\x0f"
+      branch    = "\x0305#{branch}\x0f"
+    end
+
+    # Firsts messages are for branch creation/deletion
+    if push_data["before"] =~ /^000000/
+      send_new_branch project.path_with_namespace, repo_name, committer, branch
+    elsif push_data["after"] =~ /^000000/
+      send_del_branch repo_name, committer, branch
+    end
+
+    # Next message is for number of commit pushed, if any
+    if push_data['total_commits_count'] == 0
+      return true
+    end
+
+    if push_data["before"] =~ /^000000/
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.